pandas/core/groupby/generic.py

```diff
@@ -1754,10 +1754,22 @@ def value_counts(
                     level=index_level, sort_remaining=False
                 )
 
-            if not self.as_index:
+            if self.as_index:
+                return result.__finalize__(self.obj, method="value_counts")
+            else:
                 # Convert to frame
-                result = result.reset_index(name="proportion" if normalize else "count")
-            return result.__finalize__(self.obj, method="value_counts")
+                name = "proportion" if normalize else "count"
+                index = result.index
+                columns = com.fill_missing_names(index.names)
+                if name in columns:
+                    raise ValueError(
+                        f"Column label '{name}' is duplicate of result column"
```

Comment:
why does this matter at all, simply set the columns

Reply:
I would have throught that this would invariably be the result of an error. I want value_counts to survive in the case that there were duplicate headings in the original dataframe. In this case, however, there is a clash between one of the original headings and the new heading ("count" or "proportion") being created by value_counts. Other reviewers want cases such as this to raise, since it is clearly unintentional. I will reluctantly take it out, if you insist. Please confirm.

Reply:
pls point to the test case for this

Reply:
https://github.com/johnzangwill/pandas/blob/207b55ecca5c5f81b3e0f2b337747f8880b4b45f/pandas/tests/groupby/test_frame_value_counts.py#L447